### PR TITLE
feat: completion function to MessageConsumer returns a future

### PR DESCRIPTION
**Pull request by wem**
feat: completion function to MessageConsumer returns a future

Alternative to completionHandler.

Motivation:

https://github.com/vert-x3/vertx-lang-kotlin/issues/190

### CODE REVIEW

File: src/main/java/io/vertx/core/eventbus/MessageConsumer.java [hunk 1]
```diff
@@ -90,6 +90,11 @@
    */
   void completionHandler(Handler<AsyncResult<Void>> completionHandler);
 
+  /**
+   * Like {@link #completionHandler(Handler)} but returns a {@code Future} of the asynchronous result.
+   */
+  Future<Void> completion();
```
Comment (vietj) [inline src/main/java/io/vertx/core/eventbus/MessageConsumer.java:96]: I think instead we should name it `registrationCompletion`

Comment (vietj) [reply src/main/java/io/vertx/core/eventbus/MessageConsumer.java:96]: to clarify it is related to the consumer registration.

File: src/main/java/io/vertx/core/eventbus/MessageConsumer.java [hunk 2]
```diff
@@ -90,6 +90,11 @@
    */
   void completionHandler(Handler<AsyncResult<Void>> completionHandler);
 
+  /**
+   * Like {@link #completionHandler(Handler)} but returns a {@code Future} of the asynchronous result.
+   */
+  Future<Void> registrationCompletion();
+
```
Comment (vietj) [inline src/main/java/io/vertx/core/eventbus/MessageConsumer.java:97]: we also need a `registrationCompletion(Handler<AsyncResult<Void>>)` method to go along with this

Comment (wem) [reply src/main/java/io/vertx/core/eventbus/MessageConsumer.java:97]: I think after this round i understand your intention :)

**Comment by vietj** [comment]
can you rebase on latest master ?

**Comment by vietj** [comment]
I think the main issue here is that a consumer can be registered or unregistered multiple times and it is not clear what a registration future would mean, we need to clarify this before taking any action

**Comment by tsegismont** [comment]
Superseded by #4645

**Pull request closed without merging**